packages/astroport/src/querier.rs

```diff
@@ -64,6 +61,44 @@ pub fn query_token_balance(
     Ok(resp.balance)
 }
 
+/// Returns the number of decimals that a token has.
+///
+/// * **asset_info** is an object of type [`AssetInfo`] and contains the asset details for a specific token.
+pub fn query_token_precision(
+    querier: &QuerierWrapper,
+    asset_info: &AssetInfo,
+    factory_addr: &Addr,
+) -> StdResult<u8> {
+    Ok(match asset_info {
+        AssetInfo::NativeToken { denom } => {
+            if let Some(res) = querier.query_wasm_raw(factory_addr, b"config".as_slice())? {
```

Comment:
that's copypasted from astroport...
I don't know actually why) Maybe because their factory does not guarantee that it has query config? But our factory is pretty much the only one, right?

Reply:
well i found and used already existing method that uses wasm_smart to get config